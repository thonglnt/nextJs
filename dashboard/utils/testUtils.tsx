import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '@self-types/Themes.types'

export const customRender = (children: ReactNode) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export * from '@testing-library/react'
export { customRender as render }
