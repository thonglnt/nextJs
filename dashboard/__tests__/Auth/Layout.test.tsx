// Utils
import { render } from '@utils/testUtils'

// App
import RootLayout from 'app/(auth)/layout'

describe('RootLayout', () => {
  it('should render RootLayout', () => {
    const { container } = render(<RootLayout>Content Pages</RootLayout>)

    expect(container).toMatchSnapshot()
  })
})
