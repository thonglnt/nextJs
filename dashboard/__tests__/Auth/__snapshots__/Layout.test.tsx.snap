// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout should render RootLayout 1`] = `
<div>
  <html
    lang="en"
  >
    <head>
      <meta
        charset="UTF-8"
      />
      <meta
        content="width=device-width, initial-scale=1.0"
        name="viewport"
      />
      <meta
        content="IE=edge"
        http-equiv="X-UA-Compatible"
      />
      <link
        href="/favicon.ico"
        rel="icon"
      />
    </head>
    <body>
      <main
        class="MuiBox-root css-prgmkg"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthMd css-2dgwvx-MuiContainer-root"
        >
          Content Pages
        </div>
      </main>
    </body>
  </html>
</div>
`;
